interop/tapir/src/main/scala/caliban/interop/tapir/package.scala

```diff
@@ -162,16 +162,29 @@ package object tapir {
     override protected val additionalDirectives: List[__Directive] = Nil
   }
 
-  private def extractPath[I](input: EndpointInput[I]): String =
-    input
-      .asVectorOfBasicInputs(includeAuth = false)
-      .collect {
-        case EndpointInput.FixedPath(s, _, _) => s
-      }
-      .toList match {
-      case Nil          => "root"
-      case head :: Nil  => head
-      case head :: tail => head ++ tail.map(_.capitalize).mkString
+  private def extractPath[I](endpointName: Option[String], input: EndpointInput[I]): String =
+    endpointName
+      .flatMap(toCamel)
+      .getOrElse(
+        input
+          .asVectorOfBasicInputs(includeAuth = false)
+          .collect {
+            case EndpointInput.FixedPath(s, _, _) => s
+          }
+          .toList match {
+          case Nil          => "root"
+          case head :: Nil  => head
+          case head :: tail => head ++ tail.map(_.capitalize).mkString
+        }
+      )
+
+  private def toCamel(s: String): Option[String] =
+    s.replaceAll("\\W", "_")
+      .split("_")
+      .filterNot(_.isEmpty)
+      .map(s => s"${s.head.toUpper}${s.tail.toLowerCase()}") match {
+      case Array(head, tail @ _*) => Some((head.toLowerCase ++ tail).mkString(""))
```

Comment:
If someone named it well like `addBook`, it would turn it into `addbook`, no? Maybe we should preserve casing for the tail?

Reply:
i started with
```
if (name.matches(".*\\W")) doNameFixing(name) else name
```
because overall the less a library forces naming conventions the better

it'd rather the library enable customization, and not enforce custom rules

so if a user wants `snake_case_is_superior` that should ok

What do you think?

I tent to of look at such issues from testing perspective.
Should I test the `addBook -> addbook` ? I am not convinced

Doing only `name.replaceAll("\\W","_")` could be enough.
This, of course, is only valid if the library doesn't already follow a certain style.

I am good with either case

Reply:
I agree, we probably shouldn't do too much and only prevent invalid graphql schema to be generated. So replacing the spaces with `_` is fine with me.

Reply:
so I think this is all for this PR

the second part i will have to dig myself a bit more into the lovely codebase